openslides/agenda/static/js/agenda/agenda.js

```diff
@@ -158,15 +158,64 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
         });
         return nodes;
     }
+    // open detail view link
+    $scope.openDetail = function (id) {
+        $state.go('agenda.item.detail', {id: id});
+    }
 
     // save changed item
     $scope.save = function (item) {
         Agenda.save(item);
     };
+
+    $scope.isDeleteMode = false;
+    $scope.selection = [];
+    $scope.toggleSelection = function toggleSelection(id) {
+        var index = $scope.selection.indexOf(id);
+        if ( index == -1 ) {
+            // not yet selected, add id to array
+            $scope.selection.push(id);
+        }
+        else {
+            // already selected, remove id from array
+            $scope.selection.splice(index, 1);
+        }
+    };
+    // check all checkboxes
+    $scope.checkAll = function () {
+        if ($scope.selectedAll) {
+            $scope.selectedAll = true;
+        } else {
+            $scope.selectedAll = false;
+        }
+        $scope.selection = [];
+        angular.forEach($scope.flattenedTree, function (node) {
+            node.selected = $scope.selectedAll;
+            if ($scope.selectedAll) {
+                $scope.selection.push(node.id);
+            }
+        });
+    };
+    // uncheck all checkboxes if isDeleteMode is closed
+    $scope.uncheckAll = function () {
+        if (!$scope.isDeleteMode) {
+            $scope.selection = [];
+            $scope.selectedAll = false;
+            angular.forEach($scope.flattenedTree, function (node) {
+                node.selected = false;
+            });
+        }
+    };
     // delete selected item
-    $scope.delete = function (id) {
-        Agenda.destroy(id);
+    $scope.delete = function () {
+        console.log("Delete items: " + $scope.selection);
+        $scope.selection.forEach(function(id) {
```

Comment:
When $scope.selection is not used anymore, then you can work with _.foreach(node, function (value) ...